libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -31,6 +30,12 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
 
   peer->markDagBlockAsKnown(hash);
 
+  if (!seen_dags_.insert(block.getHash())) {
+    LOG(log_dg_) << "Received dag block" << block.getHash() << " (from " << packet_data.from_node_id_.abridged()
+                 << ") already seen.";
+    return;
+  }
+
   if (dag_blk_mgr_) {
     // Do not process this block in case we already have it
     if (dag_blk_mgr_->isDagBlockKnown(block.getHash())) {
```

Comment:
so how is your new cache different to this one ? I dont like the idea having two caches that are doing basically the same thing, just checked on different places

Reply:
isDagBlockKnown saves only valid dag blocks

Reply:
we need to have cache for requested blocks and this should kind of be that, maybe I should implement it only for requested blocks.

Reply:
I see, ok... It would be nice differentiate then between these caches somehow. If you open this in 1 month and one cache is called seen_dags, another is called seen_blocks you will have no idea what is the difference between them. So at least some comment should be there on declaration and maybe the change name of the other cache so it is clear that is caches only valid_seen_blocks and this one all of the blocks...